docs/csharp/programming-guide/statements-expressions-operators/expression-bodied-members.md

```diff
@@ -0,0 +1,105 @@
+|[Property Get](#property-get-statements)  |C# 6 |
+|[Property Set](#property-set-statements)  |C# 7 |
+|[Indexer](#indexers)       |C# 7 |
+
+## Methods
+
+An expression-bodied method consists of a single expression that returns a value whose type matches the method's return type, or, for methods that return `void`, that performs some operation. For example, types that override the <xref:System.Object.ToString%2A> method typically include a single expression that returns the string representation of the current object. 
+
+The following example defines a `Person` class that overrides the <xref:System.Object.ToString%2A> method with an expression body definition. It also defines a `Show` method that displays a name to the console. Note that the `return` keyword is not used in the `ToString` expression body definition.
+
+[!code-cs[expression-bodied-methods](../../../../samples/snippets/csharp/programming-guide/classes-and-structs/expr-bodied-methods.cs)]  
+
+For more information, see [Methods (C# Programming Guide)](../classes-and-structs/methods.md).
+ 
+## Constructors
+
+An expression body definition for a constructor typically consists of a single assignment expression or a method call that handles the constructor's arguments or initializes instance state. 
+
+The following example defines a `Location` class whose constructor has a single string parameter named *name*. The expression body definition assigns the argument to the `Name` property.
+
+[!code-cs[expression-bodied-constructor](../../../../samples/snippets/csharp/programming-guide/classes-and-structs/expr-bodied-ctor.cs#1)]  
+
+For more information, see [Constructors (C# Programming Guide)](../classes-and-structs/constructors.md).
+
+## Finalizers
+
+An expression body definition for a finalizer typically contains cleanup statements, such as statements that release unmanaged resources.
+
+The following example defines a finalizer that uses an expression body definition to indicate that the finalizer has been called.
+
+[!code-cs[expression-bodied-finalizer](../../../../samples/snippets/csharp/programming-guide/classes-and-structs/expr-bodied-destructor.cs#1)]  
+
+For more information, see [Finalizers (C# Programming Guide)](../classes-and-structs/destructors.md).
+
+## Property get statements
+
+If you choose to implement a property get accessor yourself, you can use an expression body definition for single expressions that simply return the property value. Note that the `return` statement isn't used.
+
+The following example defines a `Location.Name` property whose property get accessor returns the value of the private `locationName` field that backs the property. 
+
+[!code-cs[expression-bodied-property-getter](../../../../samples/snippets/csharp/programming-guide/classes-and-structs/expr-bodied-ctor.cs#1)]  
+
+Read-only properties that use an expression body definition can be implemented without an explicit `set` statement. The syntax is:
+
+```cs
```

Comment:
i only commented once, but all places need to switch from \`\`\`cs to \`\`\`csharp